From 6c9d0601cb056f0d686391819ad6f2a4dd017acc Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Tue, 27 Sep 2022 17:52:38 +0900
Subject: [PATCH] backward compatibility: alias base::ThreadType as
 base::ThreadPriority

base::ThreadPriority is replaced by base::ThreadType in r1021517
(crrev.com/c/3329026).
base::Thread::Options::priority is renamed to thread_type. Make priority
a reference of the latter.
Add old names of enum base::ThreadPriority to corresponding entries in
new base::ThreadType enum.

Change-Id: I1738e52ad71a5aa26b70c2954d615be1e39c1399
---
 base/threading/platform_thread.h | 6 ++++++
 base/threading/thread.h          | 1 +
 2 files changed, 7 insertions(+)

diff --git a/base/threading/platform_thread.h b/base/threading/platform_thread.h
index c929b2661dc6..2eb13ae99cd8 100644
--- a/base/threading/platform_thread.h
+++ b/base/threading/platform_thread.h
@@ -95,20 +95,26 @@ enum class ThreadType : int {
   // Suitable for threads that have the least urgency and lowest priority, and
   // can be interrupted or delayed by other types.
   kBackground,
+  BACKGROUND = kBackground,
   // Default type. The thread priority or quality of service will be set to
   // platform default. In Chrome, this is suitable for handling user
   // interactions (input), only display and audio can get a higher priority.
   kDefault,
+  NORMAL = kDefault,
   // Suitable for threads which are critical to compositing the foreground
   // content.
   kCompositing,
+  DISPLAY = kCompositing,
   // Suitable for display critical threads.
   kDisplayCritical,
   // Suitable for low-latency, glitch-resistant audio.
   kRealtimeAudio,
+  REALTIME_AUDIO = kRealtimeAudio,
   kMaxValue = kRealtimeAudio,
 };
 
+using ThreadPriority = ThreadType;
+
 // Cross-platform mapping of physical thread priorities. Used by tests to verify
 // the underlying effects of SetCurrentThreadType.
 enum class ThreadPriorityForTest : int {
diff --git a/base/threading/thread.h b/base/threading/thread.h
index 167467ba0deb..c37655ef1f55 100644
--- a/base/threading/thread.h
+++ b/base/threading/thread.h
@@ -106,6 +106,7 @@ class BASE_EXPORT Thread : PlatformThread::Delegate {
 
     // Specifies the initial thread type.
     ThreadType thread_type = ThreadType::kDefault;
+    ThreadType& priority = thread_type;
 
     // If false, the thread will not be joined on destruction. This is intended
     // for threads that want TaskShutdownBehavior::CONTINUE_ON_SHUTDOWN
-- 
2.37.3.998.g577e59143f-goog

